### PR TITLE
Person instance method

### PR DESCRIPTION
**Pull request by Ambrosegithub**
###  Create opp school library app where users can borrow books for use

- Create a Person class
- Create a Student class
- Create a Teacher class 
- Inherited person object into Student AND Teacher class
- Create public method Can_use_services?
- Create a private method of_age?
- Create a play_Hooky method

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative 'person'
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, classroom = 'Unknown')
+    super(age, name)
+    @classroom = classroom
+  end
```
Comment (phelian23) [inline student.rb]: - [ ] While initializing the student class, the constructor should extends that of the person class. the name and the parent_permission should be passed when creating a student. Kindly fix that.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require_relative './person'
+class Teacher < Person
+  def initialize(specialization = 'Unknown')
+    super(age, name)
+    @specialization = specialization
```
Comment (phelian23) [inline teacher.rb]: - [ ] While initializing the Teacher class, the constructor should extends that of the person class. the name, age and the parent_permission should be passed when creating a teacher. Kindly fix that.

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @Ambrosegithub,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
No linters errors ✔️
Descriptive pull request ✔️
Descriptive commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+class Person
+  attr_accessor :name, :age, :parent_permission
+  attr_reader :id
+
+  def initialize(age, parent_permission, name = 'Unknown')
```
Comment (GzimAsani) [inline person.rb]: - [ OPTIONAL ] Add the parent_permission as default true in your initialize method  `parent_permission = true`

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**